mycpp/comparator_impls.h

```diff
@@ -0,0 +1,68 @@
+#ifndef COMPARATOR_IMPLS_H
+#define COMPARATOR_IMPLS_H
+
```

Comment:
I think this whole file can go in leaky_builtins.cc

I ran into the inline issue as well, I think that solve it